contracts/src/finance/vesting/vesting_trait.rs

```diff
@@ -0,0 +1,97 @@
+pub use pendzl::traits::Balance;
+
+use ink::primitives::AccountId;
+use pendzl::traits::Timestamp;
+#[ink::trait_definition]
+pub trait Vesting {
+    #[ink(message, payable)]
+    fn create_vest(
+        &mut self,
+        to: AccountId,
+        asset: Option<AccountId>,
+        amount: Balance,
+        vesting_start: Timestamp,
+        vesting_end: Timestamp,
+    ) -> Result<(), VestingError>;
+    #[ink(message)]
+    fn release(&mut self, asset: Option<AccountId>) -> Result<(), VestingError>;
+    #[ink(message)]
+    fn release_by_vest_id(&mut self, asset: Option<AccountId>, id: u32)
```

Comment:
consider adding "to" parameter to be able to release vest to different then caller's account